### PR TITLE
Run reconciler tests for KafkaChannel properly

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Run reconciler tests for KafkaChannel properly
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Review by pierDipi — COMMENTED** [review]
Naming e2e tests files with `e2e-*tests` has the side effect of running in the default test runner
https://github.com/knative/hack/blob/da09733f828126a171068d7ec35d06d5e65d7ae6/presubmit-tests.sh#L225
so `channel-e2e-tests.sh` will be executed during the "integration-tests" job as well.

**Comment by aliok** [comment]
What I see from that line you posted is that nothing will happen if the test file doesn't start with `e2e`

```
> find test/ ! -name "$(printf "*\n*")" -name "e2e-*tests.sh" -maxdepth 1                                                                                                                                                                                                                                            test/e2e-tests.sh
```

So, should be fine.

**Comment by aliok** [comment]
/hold

I will check each job's logs to see if correct tests are run

**Comment by aliok** [comment]
```
channel_impl.go:76: admission webhook "validation.webhook.kafka.eventing.knative.dev" denied the request: validation failed: invalid value: : spec.retentionDuration
```

**Comment by pierDipi** [comment]
> What I see from that line you posted is that nothing will happen if the test file doesn't start with `e2e`
> 
> ```
> > find test/ ! -name "$(printf "*\n*")" -name "e2e-*tests.sh" -maxdepth 1                                                                                                                                                                                                                                            
> 
> test/e2e-tests.sh
> ```
> 
> So, should be fine.

The problem is that they run twice.

**Comment by aliok** [comment]
> The problem is that they run twice.

> so channel-e2e-tests.sh will be executed during the "integration-tests" job as well.

Sorry, I still didn't get it...

`channel-e2e-tests.sh` doesn't match the `e2e-*tests.sh` pattern, so it won't be executed again.

**Comment by pierDipi** [comment]
> > The problem is that they run twice.
> 
> > so channel-e2e-tests.sh will be executed during the "integration-tests" job as well.
> 
> Sorry, I still didn't get it...
> 
> `channel-e2e-tests.sh` doesn't match the `e2e-*tests.sh` pattern, so it won't be executed again.


sorry, I thought it would match, doesn't `*` means 0 or more chars?

**Comment by aliok** [comment]
> sorry, I thought it would match, doesn't * means 0 or more chars?

Yes, but `channel-e2e-tests` starts with `channel` and not `e2e`. The beginning of it doesn't match.

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -149,6 +149,9 @@ jobs:
           export SYSTEM_NAMESPACE=knative-eventing
           export CLUSTER_DOMAIN=${CLUSTER_SUFFIX}
 
+          # Setup auth config for KafkaChannel. Uses EVENTING_KAFKA_BROKER_CHANNEL_AUTH_SCENARIO env var.
+          setup_kafka_channel_auth
```
[the remark this answers is not in the document]
Comment (aliok) [reply .github/workflows/kind-e2e.yaml]: Good point! Reverted.

**Review by pierDipi — APPROVED** [review]
/lgtm

**Comment by aliok** [comment]
/unhold

Correct jobs are run

**Pull request merged**